### PR TITLE
Feat/enforce output choice

### PR DESCRIPTION
**Pull request by paulnovello**
For baselines that extract features from internal layers, make the argument "output_layers_id" mandatory.
This is because the name and the sequencing of the layers might differ from one network to another, and setting some value by default may silently affect the results.

For instance, if the default value is -2, VIM might work for a network with `layer[-2] = flatten, layers[-1]=dense(activation=softmax)` but not with `layer[-2] = dense(activation=linear) layers[-1]=softmax`, even if it can encode the same network.

### CODE REVIEW

File: docs/notebooks/torch/demo_vim_torch.ipynb
Comment (y-prudent) [inline docs/notebooks/torch/demo_vim_torch.ipynb:1]: `output_layers_id` argument is missing in the call of VIM at cell 8

File: docs/notebooks/torch/demo_dknn_torch.ipynb
Comment (y-prudent) [inline docs/notebooks/torch/demo_dknn_torch.ipynb:1]: `output_layers_id` argument is missing in the call of DKNN at cell 5

File: docs/notebooks/tensorflow/demo_dknn_tf.ipynb
Comment (y-prudent) [inline docs/notebooks/tensorflow/demo_dknn_tf.ipynb:1]: `output_layers_id` argument is missing in the call of DKNN at cell 8

File: oodeel/methods/vim.py [hunk 1]
```diff
@@ -52,27 +52,27 @@ class VIM(OODBaseDetector):
...
     Args:
-        princ_dims: number of principal dimensions of in distribution features to
-            consider.
-            If an int, must be less than the dimension of the feature space.
-            If a float, it must be in [0,1), it represents the ratio of explained
-            variance to consider to determine the number of principal components.
-            Defaults to 0.99.
-        pca_origin: either "pseudo" for using $W^{-1}b$ where $W^{-1}$ is the pseudo
+        output_layers_id (List[Union[int, str]]): features to use for Residual score.
+            The output_id for the Energy score is set to -1 by default).
```
Comment (y-prudent) [inline oodeel/methods/vim.py]: remove line about energy scores because a little bit ambiguous

**Review by y-prudent — CHANGES_REQUESTED** [review]

**Comment by y-prudent** [comment]
LGTM!

**Review by y-prudent — APPROVED** [review]

**Review by cofri — APPROVED** [review]
